### ISSUE TITLE
Installation via snapcraft can't write to filesystem

### ISSUE THREAD

**Issue by jimafisk** [plentico/plenti#31]
Installing plenti manually by placing the binary at `/usr/local/bin/plenti` works on Ubuntu 20.04, however if instead you try using `snap install plenti` it appears that it can't write to the filesystem correctly. For example, running `plenti new site test` results in:
```
Unable to write file: open test/content/blog/_blueprint.json: no such file or directoryUnable to write file: open test/content/pages/about.json: no such file or directoryUnable to write file: open test/layout/components/template.svelte: no such file or directoryUnable to write file: open test/layout/content/404.svelte: no such file or directoryUnable to write file: open test/layout/global/head.svelte: no such file or directoryUnable to write file: open test/layout/scripts/load_component.svelte: no such file or directoryUnable to write file: open test/assets/favicon.svg: no such file or directoryUnable to write file: open test/assets/planarian.svg: no such file or directoryUnable to write file: open test/layout/scripts/sort_by_date.svelte: no such file or directoryUnable to write file: open test/content/blog/post2.json: no such file or directoryUnable to write file: open test/content/index.json: no such file or directoryUnable to write file: open test/layout/content/blog.svelte: no such file or directoryUnable to write file: open test/layout/content/index.svelte: no such file or directoryUnable to write file: open test/layout/global/footer.svelte: no such file or directoryUnable to write file: open test/layout/global/html.svelte: no such file or directoryUnable to write file: open test/content/blog/adding_pletiform.json: no such file or directoryUnable to write file: open test/content/blog/post1.json: no such file or directoryUnable to write file: open test/layout/global/nav.svelte: no such file or directoryUnable to write file: open test/content/pages/contact.json: no such file or directoryUnable to write file: open test/layout/content/pages.svelte: no such file or directoryUnable to write file: open test/layout/scripts/make_title.svelte: no such file or directoryUnable to write file: open test/plenti.json: no such file or directoryUnable to write file: open test/.gitignore: no such file or directoryUnable to write file: open test/content/pages/_blueprint.json: no such file or directoryUnable to write file: open test/layout/components/grid.svelte: no such file or directoryUnable to write file: open test/package.json: no such file or directoryCreated plenti site scaffolding in "test" folder
Installing NPM dependencies...
NPM install complete!
```

**Note**: If switching between installation methods, for example if you were to try to `snap remove plenti` then manually install you'll likely get an error like:
```
bash: /snap/bin/plenti: No such file or directory
```
To fix this you simply need to close out of your current terminal and reopen a new one.

**Comment by jimafisk** [comment]
Looks like we need to set confinement to `classic` which "allows access to your system’s resources" per: https://snapcraft.io/docs/snap-confinement

Goreleaser appears to use `strict` confinement by default: https://goreleaser.com/customization/snapcraft/

**Comment by jimafisk** [comment]
Classic confinement [needs manual review](https://forum.snapcraft.io/t/process-for-reviewing-classic-confinement-snaps/1460), so [the build](https://github.com/plentico/plenti/runs/714802775) errors with:
```
- (NEEDS REVIEW) confinement 'classic' not allowed. If your snap needs classic confinement to function, please make a request for this snap to use classic by creating a new topic in the forum using the 'store-requests' category and detail the technical reasons why classic is required.
```

**Comment by jimafisk** [comment]
The [store requests](https://forum.snapcraft.io/c/store-requests) link in the manual review process mentioned above [no longer](https://forum.snapcraft.io/t/request-for-classic-confinement-spdk-snap/17268) seems to be the correct way to prompt review.

Last build automatically created a manual review request in the plenti project dashboard: https://dashboard.snapcraft.io/snaps/plenti/revisions/15/

**Comment by jimafisk** [comment]
https://forum.snapcraft.io/t/classic-confinement-request-for-plenti/17803

**Comment by jimafisk** [comment]
Can continue using `strict` confinement, but needed to add plug to provide read / write access: https://forum.snapcraft.io/t/classic-confinement-request-for-plenti/17803/2?u=jimafisk

I had to click "reject review" at the bottom of the release 15 https://dashboard.snapcraft.io/snaps/plenti/revisions/15/. Then https://dashboard.snapcraft.io/snaps/plenti/revisions/17/ and https://dashboard.snapcraft.io/snaps/plenti/revisions/18/ had a statuses of "ready to release" so at the bottom of the page I clicked the green "Release" button for each. After that releases started working automatically when Github actions would run.

I also needed to add "network-bind" plug in order to allow local webserver to work. Previously was getting the following error:
```
listen tcp :3000: listen: operation not permitted
```
The last remain issue seems to be that the snap can't execute external scripts. This is needed in order to run `npm install` when creating a new site and to run the ejectable `build.js` which runs on every build.
